### ISSUE TITLE
ajax() returns Observable<{}> instead of Observable<AjaxResponse>

### ISSUE THREAD

**Issue by billba** [ReactiveX/rxjs#1783]
**RxJS version:** 5.0.0-beta.9

**Code to reproduce:**

```typescript
const source = Observable.ajax({ method:"GET", url: "myurl" });
```

**Expected behavior:**

Mouse over _source_ to see that it's type _Observable&lt;AjaxResponse>_

**Actual behavior:**

It's type _Observable&lt;{}>_.

**Additional information:**

What might be going on is that _AjaxObservable&lt;T>.create_ has

```typescript
      return new AjaxObservable(urlOrRequest);
```
and I think it should be:
```typescript
      return new AjaxObservable<AjaxResponse>(urlOrRequest);
```

As a workaround I was able to do:

```typescript
const source = Observable.ajax<AjaxResponse>({ method:"GET", url: "myurl" });
```

... but _ajax_ shouldn't really take a type at all and I'm not sure why it even can.

**Comment by kwonoj** [comment]
Seems it makes sense. Let me look bit more and create PR if necessary.

**Comment by kwonoj** [comment]
I've looked this into bit more to see if it's possible to get rid of generics for ajax creation method, found it's bit tricky.

> but ajax shouldn't really take a type at all and I'm not sure why it even can.

: Main reason for this is `AjaxRequest` allows `resultSelector` makes `AjaxObservable` to emit other than `AjaxResponse`, ` Observable.ajax<AjaxResponse>({ method:"GET", url: "myurl" })` is actually valid way to specify types.

While it's expected to allow generics to AjaxObservable I'm also feeling type definitions can be improved bit better to work with `resultSelector` as well as default, not sure about immediate changes though. I'll leave this open for couple of more days to get idea around.

**Comment by billba** [comment]
Ah but here's the thing -- ajax, which is to say AjaxObservable.create -- isn't written as a generic function. That's why it's so weird that you can even give it a type. I think what's happening is that TypeScript "needs" a type for that AjaxObservable constructor, and so it lets you add one farther up the stack. Which sure sounds like a bug in TypeScript, but that's a different issue for a different rep.

On the larger subject, my two cents is this: let ajax continue to return the full AjaxResponse, just like in jQuery. It shouldn't be a generic.

If you want to access a REST JSON response as a type, you don't need to resort to resultSelector. You can just do:

```typescript
const source = Observable.ajax({ method:"GET", url: "myurl" }).map(ajaxResponse => ajaxResponse.response as TheTypeOfTheRestResponse);
```

Ditto if you just want part of the response:

```typescript
const source = Observable.ajax({ method:"GET", url: "myurl" }).map(ajaxResponse => ajaxResponse.response["foo"]);
```

Both are simpler and clearer (and more Rx-y) than the equivalent resultSelector code, no?

tl;dr: let ajax be ajax

**Comment by david-driscoll** [comment]
I don't personally see a ton of value in the result selector (after looking at the code) which would make this problem very easy to solve.

I'll leave it up to @blesh and @kwonoj though.

**Comment by kwonoj** [comment]
I'm also leaning towards to drop off resultSelector, just unsure if there's solid usecases for those. (But as @billba mentioned, most cases `map` can resolve selections..)

**Comment by kwonoj** [comment]
I've filed PR #1813 to leverage as discussion point. Basically PR drops resultSelector and ensure type delivered to subscription, encourage to use map where selection needed.
